centreon/src/Centreon/Application/Controller/PlatformController.php

```diff
@@ -23,93 +23,63 @@
+use Centreon\Domain\Platform\PlatformException;
+use Centreon\Domain\PlatformInformation\Model\PlatformInformationDtoValidator;
+use Centreon\Domain\PlatformInformation\UseCase\V20\UpdatePartiallyPlatformInformation;
 use Centreon\Domain\VersionHelper;
+use Core\Common\Infrastructure\FeatureFlags;
 use FOS\RestBundle\View\View;
+use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Centreon\Domain\Platform\PlatformException;
-use Centreon\Domain\PlatformInformation\UseCase\V20\UpdatePartiallyPlatformInformation;
-use Centreon\Domain\Platform\Interfaces\PlatformServiceInterface;
-use Centreon\Domain\PlatformInformation\Model\PlatformInformationDtoValidator;
 use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
 
 /**
  * This controller is designed to manage API requests concerning the versions of the different modules, widgets on the
  * Centreon platform.
- *
- * @package Centreon\Application\Controller
  */
 class PlatformController extends AbstractController
 {
-    /**
-     * @var PlatformServiceInterface
-     */
-    private $informationService;
-
     public function __construct(
-        PlatformServiceInterface $informationService
+        private readonly PlatformServiceInterface $informationService,
+        private readonly FeatureFlags $flags
     ) {
-        $this->informationService = $informationService;
     }
 
     /**
      * Retrieves the version of modules, widgets, remote pollers from the Centreon Platform.
      *
-     * @return View
      * @throws PlatformException
+     *
+     * @return View
      */
-    public function getVersions(): View
+    public function getVersions(): Response
     {
         $webVersion = $this->informationService->getWebVersion();
         $modulesVersion = $this->informationService->getModulesVersion();
         $widgetsVersion = $this->informationService->getWidgetsVersion();
 
-        return $this->view(
+        return new JsonResponse(
             [
-                'web' => $this->extractVersion($webVersion),
-                'modules' => array_map(
-                    function ($version) {
-                        return $this->extractVersion($version);
-                    },
-                    $modulesVersion
-                ),
-                'widgets' => array_map(
-                    function ($version) {
-                        return $this->extractVersion($version);
-                    },
-                    $widgetsVersion
-                )
+                'web' => (object) $this->extractVersion($webVersion),
+                'modules' => (object) array_map($this->extractVersion(...), $modulesVersion),
+                'widgets' => (object) array_map($this->extractVersion(...), $widgetsVersion),
+                'is_cloud_platform' => $this->flags->isCloudPlatform(),
+                'feature_flags' => (object) array_filter($this->flags->getAll()),
```

Comment:
```suggestion
                'is_cloud_platform' => $this->flags->isCloudPlatform(),
```

It should be rediscussed on the process to handle correctly this needed